integration-libs/digital-payments/src/checkout/cms-components/dp-payment-method/dp-payment-callback/dp-payment-callback.component.spec.ts

```diff
@@ -0,0 +1,270 @@
+          useClass: MockDpLocalStorageService,
+        },
+        {
+          provide: GlobalMessageService,
+          useClass: GlobalMessageService,
+        },
+      ],
+    }).compileComponents();
+
+    dpPaymentService = TestBed.inject(DpCheckoutPaymentService);
+    dpStorageService = TestBed.inject(DpLocalStorageService);
+    msgService = TestBed.inject(GlobalMessageService);
+
+    spyOn(msgService, 'add').and.stub();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(DpPaymentCallbackComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+
+    spyOn(component.closeCallback, 'emit').and.callThrough();
+    spyOn(component.paymentDetailsAdded, 'emit').and.callThrough();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('ngOnInit()', () => {
+    it('should fetch payment details', () => {
+      const mockDpPaymentRequest: DpPaymentRequest = {
+        sessionId: mockSessionId,
+        signature: mockSignature,
+      };
+      spyOn(dpStorageService, 'readCardRegistrationState').and.returnValue(
+        mockDpPaymentRequest
+      );
+      spyOn(dpPaymentService, 'createPaymentDetails').and.returnValue(
+        of(mockPaymentDetails)
+      );
+
+      component.ngOnInit();
+
+      expect(dpStorageService.readCardRegistrationState).toHaveBeenCalled();
+      expect(dpPaymentService.createPaymentDetails).toHaveBeenCalledWith(
+        mockSessionId,
+        mockSignature
+      );
+    });
+
+    it('should show payment fetch error when empty payment details', () => {
+      const mockDpPaymentRequest: DpPaymentRequest = {
+        sessionId: mockSessionId,
+        signature: mockSignature,
+      };
+      spyOn(dpStorageService, 'readCardRegistrationState').and.returnValue(
+        mockDpPaymentRequest
+      );
+      spyOn(dpPaymentService, 'createPaymentDetails').and.returnValue(of({}));
+
+      component.ngOnInit();
+
+      expect(dpStorageService.readCardRegistrationState).toHaveBeenCalled();
+      expect(dpPaymentService.createPaymentDetails).toHaveBeenCalledWith(
+        mockSessionId,
+        mockSignature
+      );
+      expect(msgService.add).toHaveBeenCalledWith(
+        { key: 'dpPaymentForm.error.paymentFetch' },
+        GlobalMessageType.MSG_TYPE_ERROR
+      );
+      expect(component.closeCallback.emit).toHaveBeenCalled();
+    });
+
+    it('should show unknown error when empty payment request', () => {
+      spyOn(dpStorageService, 'readCardRegistrationState').and.returnValue({});
+
+      component.ngOnInit();
+
+      expect(dpStorageService.readCardRegistrationState).toHaveBeenCalled();
+      expect(msgService.add).toHaveBeenCalledWith(
+        { key: 'dpPaymentForm.error.unknown' },
+        GlobalMessageType.MSG_TYPE_ERROR
+      );
+      expect(component.closeCallback.emit).toHaveBeenCalled();
+    });
+    it('should show payment fetch error when correct payment details', () => {
```

Comment:
This unit test description doesn't match what it checks

Reply:
done